contracts/src/v0.8/PermissionedForwardProxy.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ConfirmedOwner.sol";
+import {getRevertMsg} from "./utils/utils.sol";
+
+/**
+ * @title PermissionedForwardProxy
+ * @notice This proxy is used to forward calls from sender to target. It maintains
+ * a permission list to check which sender is allowed to call which target
+ */
+contract PermissionedForwardProxy is ConfirmedOwner {
+  mapping(address => address) public forwardPermissionList;
```

Comment:
it will definitely make it more flexible but not sure how much gas increase is for nested maps, since we don't strictly need that flexibility for our immediate use case

Reply:
I think you're line of thinking is probably the way to go. Let's solve for our immediate needs and keep gas usage down.